internal/storage/benchmark_set_test.go

```
@@ -50,3 +51,71 @@ func BenchmarkEngineSetGetParallelNoTTL(b *testing.B) {
+// BenchmarkEngineSetVsSetFromBuffer compares the two paths head-to-head:
+// Set (makes its own buffer + copies) vs SetFromBuffer (caller owns the buffer).
+func BenchmarkEngineSetVsSetFromBuffer(b *testing.B) {
+	eng := NewEngine(1*time.Millisecond, 64, 0, nil, nil)
+	defer eng.Close()
+	val := []byte("benchmark_value_32_bytes_long_xxxxx")
+	key := "bench_key_name_typical_16"
+
+	b.Run("Set", func(b *testing.B) {
+		b.ReportAllocs()
+		for i := 0; i < b.N; i++ {
+			eng.Set(key, val, 0)
```

golangci-lint (2.12.2)

[error] Error return value of `eng.Set` is not checked

(errcheck)